transmogrifier/sources/ead.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from bs4 import Tag
+
+import transmogrifier.models as timdex
+from transmogrifier.sources.transformer import Transformer
+
+logger = logging.getLogger(__name__)
+
+
+class Ead(Transformer):
+    """EAD transformer."""
+
+    def get_optional_fields(self, xml: Tag) -> dict:
+        """
+        Retrieve optional TIMDEX fields from an EAD XML record.
+
+        Overrides metaclass get_optional_fields() method.
+
+        Args:
+            xml: A BeautifulSoup Tag representing a single EAD XML record.
+        """
+        fields: dict = {}
+
+        # alternate_titles
+
+        # If the record has more than one main title, add extras to alternate_titles
+        for index, title in enumerate(self.get_main_titles(xml)):
+            if index > 0:
+                fields.setdefault("alternate_titles", []).append(
+                    timdex.AlternateTitle(value=title)
+                )
+        return fields
+
+    @classmethod
+    def get_main_titles(cls, xml: Tag) -> list[Tag]:
+        """
+        Retrieve main title(s) from an EAD XML record.
+
+        Overrides metaclass get_main_titles() method.
+
+        Args:
+            xml: A BeautifulSoup Tag representing a single EAD XML record.
+        """
+        return [
+            ",".join(string for string in titleproper.stripped_strings)
+            for titleproper in xml.find_all("titleproper")
+        ]
+
+    @classmethod
+    def get_source_record_id(cls, xml: Tag) -> str:
+        """
+        Get the source record ID from an EAD XML record.
+
+        Overrides metaclass get_source_record_id() method.
+
+        Args:
+            xml: A BeautifulSoup Tag representing a single EAD XML record.
+        """
+        return xml.header.identifier.string.replace("oai:mit//", "")
```

Comment:
This is very specific to MIT's Archivesspace oai-pmh results. To make it somewhat more generic you could do `xml.header.identifier.string.split("//")[1]`.